### ISSUE TITLE
Server stops when attempting http connection when —daemon option specified on startup with 8.5.3 on Windows 10

### ISSUE THREAD

**Issue by jpbion** [LMS-Community/slimserver#1135]
If I specify -daemon (via startup parameter) I see this issue. If I don’t specify -daemon, I don’t. Issue quite repeatable. What I am referring to are INBOUND http connections to port 9000.

**Comment by michaelherger** [comment]
Does the process really exit? 32 or 64-bit version? Anything in server.log? Is LMS running as a service?

**Comment by jpbion** [comment]
LMS running as a service. 64 bit edition (so no taskbar icon.) when I use a utility called AlwaysUp to watch it, it reports that the process has terminated. When I don’t use it, all I see is that the windows “services” window shows it is no longer running. When I am back at my PC, I will see if server.log shows anything.Sent from my iPhoneOn Jul 16, 2024, at 8:03 AM, Michael Herger ***@***.***> wrote:﻿
Does the process really exit? 32 or 64-bit version? Anything in server.log? Is LMS running as a service?

**Comment by michaelherger** [comment]
And would it work without the `--daemon` parameter?

**Comment by jpbion** [comment]
It runs wonderfully without the daemon parameter, in the sense it survives and properly handles inbound requests to port 9000. I don’t thoroughly test it to see if everything still works when I leave that parameter off.Sent from my iPhoneOn Jul 16, 2024, at 8:25 AM, Michael Herger ***@***.***> wrote:﻿
And would it work without the --daemon parameter?

**Comment by jpbion** [comment]
The interesting thing is… the code in slimserver seems to want to say “make a daemon if not windows and the daemon parameter is given” … So it appears that windows should never make a daemon… but the code looks like it runs anyway. I was able to force daemon not happen by changingIf (!windows && daemon-flag)To always return false by adding && 0 to the Boolean expression. A kludge, but I was needing to get things working. So possibly the windows flag seems to be false under windows 10 with the 64 bit version of the app?Sent from my iPhoneOn Jul 16, 2024, at 8:27 AM, Joel Bion ***@***.***> wrote:﻿It runs wonderfully without the daemon parameter, in the sense it survives and properly handles inbound requests to port 9000. I don’t thoroughly test it to see if everything still works when I leave that parameter off.Sent from my iPhoneOn Jul 16, 2024, at 8:25 AM, Michael Herger ***@***.***> wrote:﻿
And would it work without the --daemon parameter?

**Comment by michaelherger** [comment]
Did you in any way modify how LMS is being started? There are hundreds of 64-bit Windows users of 8.5.x out there, and this is the first time I'm hearing about such a problem...

**Comment by jpbion** [comment]
No, I didn’t. I admit I am using a very old single-core CPU in a machine that is definitely not windows 11 ready. The machine is about  13 years old… I do indeed have things working now but I’ll remove my && 0 thing when I get home and check the server log to see if it reports anything. ( I do keep win 10 up to date, and LMS is the only thing running on this machine)Sent from my iPhoneOn Jul 16, 2024, at 8:50 AM, Michael Herger ***@***.***> wrote:﻿
Did you in any way modify how LMS is being started? There are hundreds of 64-bit Windows users of 8.5.x out there, and this is the first time I'm hearing about such a problem...

**Comment by jpbion** [comment]
Here is the exception in the log file, with the exception happening as soon as I try to go from a browser to localhost:9000.

 

It may be that for some reason the local (in-Windows) firewall is BLOCKING communication from this device to itself, so I am looking into that. Weird that it works when it’s not a daemon, though…

 

…and weird that LMS would throw an *exception* for having a target refusing a connection.

 

[24-07-15 20:36:04.7547] Slim::Schema::Storage::throw_exception (121) Error: IO::Socket::INET: connect: No connection could be made because the target machine actively refused it.

[24-07-15 20:36:04.7550] Slim::Schema::Storagre::throw_exception (121) Backtrace:

 

   frame 0: Slim::Utils::Log::logBacktrace (C:/Program Files/Squeezebox/server/Slim/Schema/Storage.pm line 121)

   frame 1: Slim::Schema::Storage::throw_exception (C:/Program Files/Squeezebox/server/Slim/Schema/Storage.pm line 46)

   frame 2: Slim::Schema::Storage::dbh (C:/Program Files/Squeezebox/server/Slim/Schema.pm line 2249)

 

frame 3: Slim::Schema::forceCommit (C:\Program Files\Squeezebox\server\slimserver.pl line 1169)

   frame 4: main::cleanup (C:\Program Files\Squeezebox\server\slimserver.pl line 1149)

   frame 5: main::stopServer (C:\Program Files\Squeezebox\server\slimserver.pl line 678)

   frame 6: main::main (C:\Program Files\Squeezebox\server\slimserver.pl line 1214)
